redisdb/datadog_checks/redisdb/redisdb.py

```diff
@@ -292,22 +293,55 @@ def _check_key_lengths(self, conn, instance, tags):
                         continue
 
                     if key_type == 'list':
-                        lengths[text_key] += db_conn.llen(key)
+                        keylen = db_conn.llen(key)
+                        lengths[text_key]["length"] += keylen
+                        lengths_overall[text_key] += keylen
                     elif key_type == 'set':
-                        lengths[text_key] += db_conn.scard(key)
+                        keylen = db_conn.scard(key)
+                        lengths[text_key]["length"] += keylen
+                        lengths_overall[text_key] += keylen
                     elif key_type == 'zset':
-                        lengths[text_key] += db_conn.zcard(key)
+                        keylen = db_conn.zcard(key)
+                        lengths[text_key]["length"] += keylen
+                        lengths_overall[text_key] += keylen
                     elif key_type == 'hash':
-                        lengths[text_key] += db_conn.hlen(key)
+                        keylen = db_conn.hlen(key)
+                        lengths[text_key]["length"] += keylen
+                        lengths_overall[text_key] += keylen
+                    elif key_type == 'string':
+                        # Send 1 if the key exists as a string
+                        lengths[text_key]["length"] += 1
+                        lengths_overall[text_key] += 1
                     else:
                         # If the type is unknown, it might be because the key doesn't exist,
                         # which can be because the list is empty. So always send 0 in that case.
-                        lengths[text_key] += 0
-
-        # send the metrics
-        for key, total in iteritems(lengths):
-            self.gauge('redis.key.length', total, tags=tags + ['key:' + key])
+                        lengths[text_key]["length"] += 0
+                        lengths_overall[text_key] += 0
+
+                    # Tagging with key_type since the same key
+                    #   can exist with  different key_type per db
+                    lengths[text_key]["key_type"] = key_type
+
+            # Send the metrics for each db in the redis instance.
+            for key, total in iteritems(lengths):
+                # Only send non-zeros if tagged per db.
+                if total["length"] > 0:
+                    self.gauge(
+                        'redis.key.length',
+                        total["length"],
+                        tags=tags + [
+                            'key:' + key,
```

Comment:
We tend to favor the use of the `.format()` fonction for this.

Reply:
not sure what you mean. where should i use the format() function? can you link me to an example from other integration perhaps?

Reply:
used the format function in setting the tags assuming I understood correctly.